test/cypress/integration/payment-links/manage_payment_links_spec.js

```diff
@@ -0,0 +1,194 @@
+    const products = [
+      buildPaymentLinkOpts('Pay for a parking permit', 'example.com/my-service/pay-for-a-parking-permit', 'en'),
+      buildPaymentLinkOpts('Pay for a firearm', 'example.com/my-service/pay-for-a-firearm', 'en'),
+      buildPaymentLinkOpts('Talu am drwydded barcio', 'example.com/my-service/talu-am-drwydded-barcio', 'cy'),
+      buildPaymentLinkOpts('Talu am arf tanio', 'example.com/my-service/talu-am-arf-tanio', 'cy')
+    ]
+
+    beforeEach(() => {
+      cy.task('setupStubs', [
+        commonStubs.getUserStub(userExternalId, [gatewayAccountId]),
+        commonStubs.getGatewayAccountStub(gatewayAccountId, 'test', 'worldpay'),
+        getProductsStub(products)
+      ])
+    })
+
+    it('should list English payment links and Welsh payment links separately', () => {
+      cy.visit('/create-payment-link/manage')
+
+      cy.get('h1').should('contain', 'Manage payment links')
+      cy.get('.payment-links-list--header').should('contain',
+        'There are 4 payment links')
+
+      cy.get('ul.payment-links-list').eq(0).find('li').should('have.length', 2)
+      cy.get('ul.payment-links-list').eq(0).find('li').eq(0).within(() => {
+        cy.get('h2').contains(products[0].name)
+        cy.get('a').contains(products[0].links[0].href)
+      })
+      cy.get('ul.payment-links-list').eq(0).find('li').eq(1).within(() => {
+        cy.get('h2').contains(products[1].name)
+        cy.get('a').contains(products[1].links[0].href)
+      })
+
+      cy.get('div#welsh-payment-links').should('exist').within(() => {
+        cy.get('h3').contains('Welsh payment links').should('exist')
+
+        cy.get('ul.payment-links-list').find('li').should('have.length', 2)
+        cy.get('ul.payment-links-list').find('li').eq(0).within(() => {
+          cy.get('h2').contains(products[2].name)
+          cy.get('a').contains(products[2].links[0].href)
+        })
+        cy.get('ul.payment-links-list').find('li').eq(1).within(() => {
+          cy.get('h2').contains(products[3].name)
+          cy.get('a').contains(products[3].links[0].href)
+        })
+      })
+    })
+  })
+
+  describe('Only Welsh payment links', () => {
+    const products = [
+      buildPaymentLinkOpts('Talu am drwydded barcio', 'example.com/my-service/talu-am-drwydded-barcio', 'cy'),
+      buildPaymentLinkOpts('Talu am arf tanio', 'example.com/my-service/talu-am-arf-tanio', 'cy')
+    ]
+
+    beforeEach(() => {
+      cy.task('setupStubs', [
+        commonStubs.getUserStub(userExternalId, [gatewayAccountId]),
+        commonStubs.getGatewayAccountStub(gatewayAccountId, 'test', 'worldpay'),
+        getProductsStub(products)
+      ])
+    })
+
+    it('should list Welsh payment links', () => {
+      cy.visit('/create-payment-link/manage')
+
+      cy.get('h1').should('contain', 'Manage payment links')
+      cy.get('.payment-links-list--header').should('contain',
+        'There are 2 payment links')
+
```

Comment:
could there be a way to test there are no english payment links here?

Reply:
good point, I could check that there is only one "ul.payment-links-list"

Reply:
why one and not zero?

Reply:
one because there is a list of Welsh payment links within 'div#welsh-payment-links' which will be picked up by that selector.